Client/Writes/PointData.Builder.cs

```diff
@@ -0,0 +1,300 @@
+                    case WritePrecision.Us:
+                        time = (BigInteger)(timestamp.Ticks * 0.1);
+                        break;
+                    case WritePrecision.Ms:
+                        time = (BigInteger)timestamp.TotalMilliseconds;
+                        break;
+                    case WritePrecision.S:
+                        time = (BigInteger)timestamp.TotalSeconds;
+                        break;
+                }
+
+                _time = time;
+                _precision = timeUnit;
+                return this;
+            }
+
+            /// <summary>
+            /// Updates the timestamp for the point represented by <see cref="DateTime"/>.
+            /// </summary>
+            /// <param name="timestamp">the timestamp</param>
+            /// <param name="timeUnit">the timestamp precision</param>
+            /// <returns></returns>
+            public Builder Timestamp(DateTime timestamp, WritePrecision timeUnit)
+            {
+                if (timestamp != null && timestamp.Kind != DateTimeKind.Utc)
+                {
+                    throw new ArgumentException("Timestamps must be specified as UTC", nameof(timestamp));
+                }
+
+                var timeSpan = timestamp.Subtract(EpochStart);
+
+                return Timestamp(timeSpan, timeUnit);
+            }
+
+            /// <summary>
+            /// Updates the timestamp for the point represented by <see cref="DateTime"/>.
+            /// </summary>
+            /// <param name="timestamp">the timestamp</param>
+            /// <param name="timeUnit">the timestamp precision</param>
+            /// <returns></returns>
+            public Builder Timestamp(DateTimeOffset timestamp, WritePrecision timeUnit)
+            {
+                return Timestamp(timestamp.UtcDateTime, timeUnit);
+            }
+
+            /// <summary>
+            /// Updates the timestamp for the point represented by <see cref="Instant"/>.
+            /// </summary>
+            /// <param name="timestamp">the timestamp</param>
+            /// <param name="timeUnit">the timestamp precision</param>
+            /// <returns></returns>
+            public Builder Timestamp(Instant timestamp, WritePrecision timeUnit)
+            {
+                BigInteger time;
+                switch (timeUnit)
+                {
+                    case WritePrecision.S:
+                        time = timestamp.ToUnixTimeSeconds();
+                        break;
+                    case WritePrecision.Ms:
+                        time = timestamp.ToUnixTimeMilliseconds();
+                        break;
+                    case WritePrecision.Us:
+                        time = (long)(timestamp.ToUnixTimeTicks() * 0.1);
+                        break;
+                    default:
+                        time = (timestamp - NodaConstants.UnixEpoch).ToBigIntegerNanoseconds();
+                        break;
+                }

```

Comment:
Could you please reuse an implementation from `PointData`? 

The future maintains will be much easier. Feel free to create something like:

```c#
internal static BigInteger? InstantToBigInteger(Instant timestamp, WritePrecision timeUnit)
{
    ...
}
```

Reply:
Done, see previous point